commit 131d8cebad66c219f36a63e47d21d7840c478ed7 (refs/changes/16/3216/6)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-16T20:18:23+01:00 (1 year, 9 months ago)
    
    TOOLS-1974 sdc-scripts refactor sdc_common_setup
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/util.sh b/lib/util.sh
index 8e07285..3cb8881 100644
--- a/lib/util.sh
+++ b/lib/util.sh
@@ -694,21 +694,18 @@ function sdc_common_setup
     _sdc_log_rotation_setup
     _sdc_mdata_rbac_setup
 
-    if [[ ! -f /var/svc/setup_complete ]]; then
-        if [[ ${ZONE_ROLE} != "assets" ]]; then
-            if [[ ${ZONE_ROLE} == "sapi" && "${SAPI_PROTO_MODE}" == \
-              "true" ]]; then
-                echo "Skipping config-agent/SAPI instance setup: 'sapi' " \
-                  "zone in proto mode" >&2
-            else
-                setup_config_agent
-                download_metadata
-                write_initial_config
-                registrar_setup
-            fi
-        fi
+    if [[ -f /var/svc/setup_complete ]]; then
+        echo "Skip config-agent/registrar setup: zone already setup" >&2
+    elif [[ ${ZONE_ROLE} == "assets" ]]; then
+        echo "Skip config-agent/registrar setup: assets zone" >&2
+    elif [[ ${ZONE_ROLE} == "sapi" && "${SAPI_PROTO_MODE}" == "true" ]]; then
+        echo "Skip config-agent/registrar setup: sapi zone in proto mode" >&2
     else
-        echo "Already setup, skipping SAPI and registrar initialization." >&2
+        echo "Setup config-agent/registrar" >&2
+        setup_config_agent
+        download_metadata
+        write_initial_config
+        registrar_setup
     fi
 
     _sdc_enable_cron
